### PR TITLE
Fix review issues for adding into rancher charts

### PR DESCRIPTION
**Pull request by yaocw2020**
Related pr: https://github.com/rancher/charts/pull/1595

### CODE REVIEW

File: charts/harvester-cloud-provider/Chart.yaml [hunk 1]
```diff
@@ -11,11 +11,13 @@ description: A Helm chart for Harvester Cloud Provider
 # a dependency of application charts to inject those utilities and functions into the rendering
 # pipeline. Library charts do not define any templates and therefore cannot be deployed.
 type: application
+keywords:
+  - Infrastructure
```
Comment (guangbochen) [inline charts/harvester-cloud-provider/Chart.yaml:15]: add `harvester` as well, thanks

File: charts/harvester-csi-driver/Chart.yaml [hunk 2]
```diff
@@ -11,11 +11,13 @@ description: A Helm chart for Harvester CSI driver
 # a dependency of application charts to inject those utilities and functions into the rendering
 # pipeline. Library charts do not define any templates and therefore cannot be deployed.
 type: application
+keywords:
+  - Infrastructure
```
Comment (guangbochen) [inline charts/harvester-csi-driver/Chart.yaml:15]: ditto

File: charts/harvester-cloud-provider/Chart.yaml [hunk 3]
```diff
@@ -31,6 +34,13 @@ annotations:
   catalog.cattle.io/rancher-version: ">= 2.6.1"
   catalog.cattle.io/ui-component: harvester-cloud-provider
   catalog.cattle.io/display-name: Harvester Cloud Provider
+  # A SemVer range of compatible Kubernetes versions, just like Helm's kubeVersion.
+  # The difference is that ours will only hide the chart in the dashboard UI, and still
+  # allow users to edit (helm upgrade) the chart; the helm's kubeVersion disables edit in installed charts.
+  catalog.cattle.io/kube-version: "> 18.0.0"
```
Comment (cbron) [inline charts/harvester-cloud-provider/Chart.yaml:40]: I'm assuming this supports 1.22 then.

File: charts/harvester-cloud-provider/values.yaml [hunk 4]
```diff
@@ -51,8 +49,13 @@ tolerations:
 - effect: NoExecute
```
Comment (cbron) [inline charts/harvester-cloud-provider/values.yaml:49]: Few lines above, still needs nodeSelector to pick linux: https://github.com/rancher/charts/blob/dev-v2.6/charts/rancher-logging/rancher-logging/100.0.1%2Bup3.15.0/values.yaml#L41-L42

**Review by cbron — COMMENTED** [review]
Still needs node selector, then looks good

File: charts/harvester-cloud-provider/values.yaml [hunk 5]
```diff
@@ -11,15 +11,13 @@ image:
   tag: v0.1.1
 
 clusterName: ""
-cloudConfigPath: "/var/lib/rancher/rke2/etc/config-files/cloud-provider-config"
+cloudConfigPath: "/etc/kubernetes/cloud-config"
```
Comment (gitlawr) [inline charts/harvester-cloud-provider/values.yaml:14]: I suggest we use the rke2 path as the default

File: charts/harvester-cloud-provider/Chart.yaml (hunk 3 above)
Comment (yaocw2020) [reply charts/harvester-cloud-provider/Chart.yaml:40]: Support RKE1 >= v1.3.3

File: charts/harvester-cloud-provider/values.yaml (hunk 5 above)
Comment (yaocw2020) [reply charts/harvester-cloud-provider/values.yaml]: Set the rke1 path as the default and add the questions.yml

**Review by gitlawr — APPROVED** [review]
LGTM

**Review by guangbochen — APPROVED** [review]

**Review by cbron — APPROVED** [review]

**Comment by guangbochen** [comment]
@yaocw2020 can u please help to update the PR of https://github.com/rancher/charts/pull/1595 and backport the changes to the `master` branch, thanks.

**Comment by yaocw2020** [comment]
> @yaocw2020 can u please help to update the PR of [rancher/charts#1595](https://github.com/rancher/charts/pull/1595) and backport the changes to the `master` branch, thanks.

Updated

**Pull request merged**